docs: Refactor for Dasharo Entry Subscription nomenclature (#634)
Dasharo/docs: docs/dev-proc/versioning.md

and [Keep A Changelog](https://keepachangelog.com/en/1.0.0/) to document
changes introduced in new releases.

Major version zero (0.y.z) is for initial development or first release issued
and may not support all Dasharo Quality Criteria.

The only way to map Dasharo Version to version of Open Source Firmware
framework or other components included in Dasharo Release is through release
notes. Link to Dasharo Release Notes for your hardware platform can be found
in menu on the left side (Supported Hardware->Hardware Model->Releases).

## Dasharo Entry Subscription releases

> Dasharo Supporters receive firmware updates more frequently than the

Reviewer: ```suggestion
Dasharo Entry Subscribers receive firmware updates more frequently than the
```

community. Number of updates per year depends on the number of Dasharo
Subscriptions sold and the availability of other funding (e.g., NLNet,

> corporate sponsors, community donations) but is less than 2 updates per year.

Reviewer: This sounds weird. Dasharo Community Releases I agree, there would be less than two. But we should have no such arbitrary limit for Dasharo Entry Subscription Releases (previously Dasharo Supporters Releases).

Author: I haven't changed the paragraph except the nomenclature (but it got rewrapped and GH detected it as an edition of whole paragraph). I think it was meant to be `no less than 2 updates per year` for subscribers @pietrushnic

Reviewer: @miczyg1, please fix it, since we have occasion to get rid of such minor typos.

Author: @pietrushnic I would actually get rid of that part. How can we guarantee there is enough subscription bought to provide two releases per year? And how long can we guarantee such schedule? Sounds like TOS

Reviewer: @miczyg1 you are right, but we should explain difference between Dasharo Community Release and Dasharo Entry Subscribers Release.

Author: The first sentence already says:

Dasharo ~~Supporters~~ Entry Subscribers receive firmware updates more frequently than the community

So I think this change should suffice @pietrushnic :

```suggestion
corporate sponsors, community donations).
```

Author: Then we also have:

```
Fixes and features introduced in Dasharo Entry Subscription
releases will also be available later as community releases with public
pre-built binaries in the respective release pages. In short, being a Dasharo
Supporter gives an early access to the newest features and fixes.
```

Reviewer: Terminology is not consistent. We should have finished a number of terms, and now we have DES, community release, and Dasharo Supporter. Some are written with big letters like legal/marketing names. Some are small, and I have no idea how to associate those.

During DUGs, so for the last five months, we have tried to use just two terms regarding releases:
- Dasharo Community Release
- Dasharo Entry Subscription Release (previous Dasharo Supporters Release for Dasharo Support Entrance Subscribers)

According to the above, we may have:
- Dasharo Community Release subscribers - those who subscribed to free of charge mailing list on docs.dashro.com
- Dasharo Entry Subscription Release subscribers (:man_facepalming:) - those who paid for DES or received it because of other reasons

The language used in this paragraph may mean the same or may mean something different. Consistency in the language is very important if we don't want to have endless support discussions and, in case of questions, send reliable sources of answers.

Author: I hope I applied consistent nomenclature everywhere @pietrushnic ccddb79e6f66511cec75e029aebb31490ae56b77

Dasharo Entry Subscription releases are characterized by a changing patch
version (`z`). Fixes and features introduced in Dasharo Entry Subscription

> releases will also be available later as community releases with public

Reviewer: ```suggestion
releases will also be available later as Dasharo Community Releases with public
```

pre-built binaries in the respective release pages. In short, being a Dasharo

> Supporter gives an early access to the newest features and fixes.

Reviewer: ```suggestion
Subscriber gives early access to the newest features and fixes.
```

> [How to become Dasharo Supporter?](../ways-you-can-help-us.md#become-a-dasharo-supporter)

Reviewer: ```suggestion
[How to become Dasharo Entry Subscriber?](../ways-you-can-help-us.md#become-a-dasharo-supporter)
```
